Dockerfile

```
@@ -0,0 +1,196 @@
+    sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-9 90 \
+        --slave /usr/bin/g++ g++ /usr/bin/g++-9 \
+        --slave /usr/bin/gcov gcov /usr/bin/gcov-9 && \
+    curl -sSLo bazel https://github.com/bazelbuild/bazelisk/releases/download/v1.18.0/bazelisk-linux-${ARCH} && \
+    sudo install bazel /usr/local/bin && \
+    ln -s /mnt/bazel-cache ~/.cache/bazel
+
+# Install npm
+# https://docs.ray.io/en/releases-2.5.1/ray-contribute/development.html#preparing-to-build-ray-on-linux
+COPY --from=node:14-bullseye /usr/local/lib/node_modules /usr/local/lib/node_modules
+COPY --from=node:14-bullseye /usr/local/bin/node /usr/local/bin/
+RUN sudo ln -s ../lib/node_modules/npm/bin/npm-cli.js /usr/local/bin/npm
+RUN node --version && \
+    npm --version
+
+ENV JULIA_PROJECT /Ray.jl
+ARG JLL_JULIA_PROJECT=${JULIA_PROJECT}/ray_julia_jll
+RUN sudo mkdir -p ${JULIA_PROJECT} && \
+    sudo chown ray ${JULIA_PROJECT}
+
+# Install custom Ray CLI which supports the Julia language.
+# https://docs.ray.io/en/releases-2.5.1/ray-contribute/development.html#building-ray-on-linux-macos-full
+ARG RAY_ROOT=/ray
+ARG RAY_COMMIT=448a83caf4
+ARG RAY_GEN_CACHE_DIR=/mnt/ray-generated
+ARG RAY_CACHE_CLEAR=false
+RUN sudo mkdir -p ${RAY_ROOT} && \
+    sudo chown ray ${RAY_ROOT}
+RUN --mount=type=cache,sharing=locked,target=/mnt/bazel-cache,uid=1000,gid=100 \
+    --mount=type=cache,sharing=locked,target=${RAY_GEN_CACHE_DIR},uid=1000,gid=100 \
+    set -eux && \
+    git clone https://github.com/beacon-biosignals/ray ${RAY_ROOT} && \
+    git -C ${RAY_ROOT} checkout ${RAY_COMMIT} && \
+    mkdir -p ${JLL_JULIA_PROJECT}/deps && \
+    ln -s ${RAY_ROOT} ${JLL_JULIA_PROJECT}/deps/ray && \
+    cd ${JLL_JULIA_PROJECT}/deps/ray && \
+    #
+    # Allow builders to clear just the Ray CLI cache.
+    if [ "${RAY_CACHE_CLEAR}" != "false" ]; then \
+        bazel clean --expunge && \
+        rm -rf ${RAY_GEN_CACHE_DIR}/*; \
+    fi && \
+    #
+    # The Ray `BUILD.bazel` includes a bunch of `copy_to_workspace` rules which copy build output
+    # into the Ray worktree. When we only restore the Bazel cache then re-building causes these
+    # rules to be skipped resulting in `error: [Errno 2] No such file or directory`. By manually
+    # saving/restoring these files we can work around this.
+    if [ -d ${RAY_GEN_CACHE_DIR}/.git ]; then \
+        cd ${RAY_GEN_CACHE_DIR} && \
+        cp -rp --parents \
+            python/ray/_raylet.so \
+            python/ray/core/generated \
+            python/ray/serve/generated \
+            python/ray/core/src/ray/raylet/raylet \
+            python/ray/core/src/ray/gcs \
+            ${RAY_ROOT} && \
+        cd -; \
+    fi && \
+    #
+    # Build the dashboard
+    cd ${JLL_JULIA_PROJECT}/deps/ray/python/ray/dashboard/client && \
+    npm ci && \
+    npm run build && \
+    #
+    # Build Ray for Python
+    cd ${JLL_JULIA_PROJECT}/deps/ray/python && \
+    pip install --verbose . && \
```

I needed to install the additioanl "default" dependencies in order to be able to use things like `ray list objects`...is there a reason not to do that here?

```suggestion
    pip install --verbose ".[default]" && \
```

---

It doesn't appear to be required to run jobs on the cluster. Possibly it would be useful if you are using the Ray CLI from within the cluster but this will take additional time to build and I doubt we'll use these features.